### PR TITLE
Fixes rifle scopes working despite not having the gun in hand

### LINKED ISSUE #56898: scope zoom doesn't check that gun is in hand
https://github.com/tgstation/tgstation/blob/42ef7decf8e0b7843dfced0eac753ba47add6a4d/code/modules/projectiles/gun.dm#L671

image

### PR DESCRIPTION
**Pull request by RandomGamer123**
## About The Pull Request
Gun scopes no longer work unless you are holding them in hand (they persist even if you switch your active hand though, useful for reloading)

## Why It's Good For The Game
Fixes #56898

## Changelog
:cl:
fix: Gun scopes no longer work if you don't have the gun in your hands
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -676,10 +676,14 @@
 	var/obj/item/gun/gun = null
 
 /datum/action/toggle_scope_zoom/Trigger()
-	gun.zoom(owner, owner.dir)
+	. = ..()
+	if(.)
+		gun.zoom(owner, owner.dir)
```
Comment (Rohesie) [inline code/modules/projectiles/gun.dm:681]:
```suggestion
	if(!.)
		return
	gun.zoom(owner, owner.dir)
```
Early return.

**Comment by Rohesie** [comment]
This won't be enough. Adding the action in `/obj/item/gun/pickup(mob/user)` is a terrible choice.
We already have an item action system that gets added through `equipped()` and we should be using like that.

**Comment by RandomGamer123** [comment]
> This won't be enough. Adding the action in `/obj/item/gun/pickup(mob/user)` is a terrible choice.
> We already have an item action system that gets added through `equipped()` and we should be using like that.

I did some testing before this and the scope gets removed automatically when you drop it or move it to like your back. For the equipped() thing i'll look into it

**Comment by Rohesie** [comment]
The `if(owner.get_active_held_item() != gun)` part is a hacky workaround for hacky code.
Then again, it's not your code and a fix's a fix.
But in the long term we'll want to get this sorted properly.

**Pull request merged**